server/worker/worker.go

```diff
@@ -0,0 +1,338 @@
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+
+	return req, nil
+}
+
+func (api *Client) Get(endpoint string) ([]byte, error) {
+	req, err := api.prepare("GET", endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+	}
+
+	defer res.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) Post(endpoint string, payload []byte) ([]byte, error) {
+	req, err := api.prepare("POST", endpoint, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+	}
+	defer res.Body.Close()
+
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) GetDashboard(uid string) *Dashboard {
+	res, err := api.Get("/api/dashboards/uid/" + uid)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - api.Get failed",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	var dashboardData map[string]interface{}
+	err = json.Unmarshal(res, &dashboardData)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - Failed to parse dashboardData response",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	return &Dashboard{
+		Uid:  uid,
+		Data: dashboardData,
+	}
+}
+
+func (api *Client) updateDashboards(dashboards []Dashboard) {
+	for _, dashboard := range dashboards {
+		if dashboard.Title == "test dashboard - empty" {
+			dashboard.Data["overwrite"] = true
+			dashboard.Data["message"] = "PRO - Auto-add analytics panel"
+
+			payload, err := json.Marshal(dashboard.Data)
+			if err != nil {
+				return
+			}
+
+			res, err := api.Post("/api/dashboards/db", payload)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to update dashboard",
+					"error", err,
+				)
+
+				return
+			}
+
+			var responseAsStruct DashboardUpdateResponse
+			err = json.Unmarshal(res, &responseAsStruct)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to parse update response",
+					"error", err,
+				)
+
+				return
+			}
+
+			if responseAsStruct.Status == "success" {
+				level.Info(api.Logger).Log(
+					"status", "success",
+					"message", "Added analytics to "+dashboard.Title,
+					"error", err,
+				)
+			}
+		}
+	}
+}
+
+func (api *Client) getAnalyticsPanelData(panelId int) map[string]interface{} {
```

Comment:
`get` prefix is misleading - it creates panel configuration, instead of getting/loading it via the api, as you would incorrectly expect when looking code where it is called from